app/views/ExposureHistory/DateInfoHeader.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useEffect,
+  useContext,
+} from 'react';
+import { StyleSheet } from 'react-native';
+import { useTranslation } from 'react-i18next';
+import dayjs, { Dayjs } from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+dayjs.extend(relativeTime);
+
+import { Typography } from '../../components/Typography';
+import { Typography as TypographyStyles } from '../../styles';
+import { fetchLastExposureDetectionDate } from '../../bt/nativeModule';
+import ExposureHistoryContext from '../../ExposureHistoryContext';
+
+const DateInfoHeader: FunctionComponent = () => {
+  const [
+    lastExposureDetectionDate,
+    setLastExposureDetectionDate,
+  ] = useState<Dayjs | null>(null);
+  const { t } = useTranslation();
+  const { exposureHistory } = useContext(ExposureHistoryContext);
+
+  useEffect(() => {
+    fetchLastExposureDetectionDate().then((exposureDetectionDate) => {
```

Comment:
I tend to try to keep IO as close to the edge of the system as possible as well as keeping the view layer and data layer independent, i.e. functional core / imperative shell. What do you think about having this call happen in the ExposureHistoryContext?  And having the view component consume this data from the context?

Reply:
I like this idea, however, if we separate this to the context we are potentially running a process that gets info from the native side surrounding a number of elements that potentially don't interact with it. So looking it from an encapsulation perspective, if we put this in the context it feels to me that we are saying, `this piece is necessary for the functioning of this block` when in reality we could live without it. Not feeling strongly either way just curious about how the encapsulation stands off with the data vs view layer organization.

Reply:
Ended up moving it to the context, let me know what you think 👍